### PR TITLE
feat(regional 🇮🇳): auto-generate e-Way Bill JSON from Sales Invoice

### PR DESCRIPTION
**Pull request by sagarvora**
**Requires** https://github.com/frappe/frappe/pull/7066 (merged)
Documentation PR: https://github.com/frappe/erpnext_com/pull/280

---

### Screenshots

**Transporter Info** section in Delivery Note (order changed):

Screenshot from 2019-03-13 00-04-05

**e-Way Bill No.** field in Sales Invoice (allowed on submit):

Screenshot from 2019-03-13 00-02-44

**Generate e-Way Bill JSON** button in Sales Invoice:

Screenshot from 2019-03-13 00-02-56

**Transporter Info** section in Sales Invoice (custom fields in regional module):

Screenshot from 2019-03-13 00-05-08

**Bulk Action** for generating e-Way Bills from List View:

Screenshot from 2019-03-13 00-05-33

---

E-waybill JSON Spec and Example (latest):
https://docs.ewaybillgst.gov.in/Documents/bulkewb/EWB_Attributes1611.xlsx

Example Output JSON File:
https://jsonbin.io/5c880189adeb832d3ec6e5cf

Tested working on ewaybillgst.gov.in

### CODE REVIEW

**Comment by sagarvora** [comment]
Documentation PR: https://github.com/frappe/erpnext_com/pull/280

File: erpnext/regional/india/utils.py [hunk 1]
```diff
@@ -234,3 +240,238 @@ def calculate_hra_exemption_for_period(doc):
 		exemptions["monthly_house_rent"] = monthly_rent
 		exemptions["total_eligible_hra_exemption"] = eligible_hra
 		return exemptions
+
+
+@frappe.whitelist()
+def generate_ewb_json(dt, dn):
```
Comment (nabinhait) [inline erpnext/regional/india/utils.py]: split into smaller function

File: erpnext/accounts/doctype/sales_invoice/regional/india.js [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+				}
+			};
+		});
+	},
+
+	refresh: function(frm) {
+		if(frm.doc.docstatus == 1 && !frm.is_dirty()
+			&& !frm.doc.is_return && !frm.doc.ewaybill) {
+
+			frm.add_custom_button('Generate e-Way Bill JSON', () => {
```
Comment (nabinhait) [inline erpnext/accounts/doctype/sales_invoice/regional/india.js]: Put it under "Make" button

**Review by nabinhait — CHANGES_REQUESTED** [review]
- Write a patch to make new custom fields
- Test cases

File: erpnext/accounts/doctype/sales_invoice/regional/india.js (hunk 2 above)
Comment (deepeshgarg007) [reply erpnext/accounts/doctype/sales_invoice/regional/india.js]: Fixed

**Review by nabinhait — APPROVED** [review]

**Pull request merged**